batchflow/opensets/base.py

```diff
@@ -27,13 +27,12 @@ def download(self, path):
         _ = path
         return None
 
-    def split(self, shares=0.8, shuffle=False):
-        if self.train_test:
-            train_data, test_data = self.preloaded  # pylint:disable=unpacking-non-sequence
-            self.train = type(self)(self._train_index, self.batch_class, train_test=False, preloaded=train_data)
-            self.test = type(self)(self._test_index, self.batch_class, train_test=False, preloaded=test_data)
-        else:
-            super().split(shares, shuffle)
+    def _infer_train_test_index(self, train_len, test_len):
+        total_len = train_len + test_len
+        index = DatasetIndex(list(range(total_len)))
```

Comment:
As far as I know, `DatasetIndex` can be initialized with an integer with the same result